### PR TITLE
Add Crust::Middleware::Directory

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: lib/Crust/App/Directory.pm6 [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+method allow-path-info() { False }
+
+method serve-dir(Hash $env, Str $dir) {
+    my $path = $env<PATH_INFO> || '';
+    return
+        301,
+        [
+			'Content-Type' => "text/plain",
```
Comment (tokuhirom) [inline lib/Crust/App/Directory.pm6]: mixed indent.
are you emacs user?

**Comment by tokuhirom** [comment]
:+1:

**Comment by tokuhirom** [comment]
ah, could you add test case?

**Comment by mattn** [comment]
will do

File: lib/Crust/App/Directory.pm6 [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+method call(Hash $env) {
+    my ($file, $path-info, $error-res) = $!dir || self.locate-file($env);
+    return |$error-res if $error-res;
+
+    return self.serve-path($env, $file) if $file.IO.f;
+    return self.serve-dir($env, $file);
+}
+
+method allow-path-info() { False }
```
Comment (tokuhirom) [inline lib/Crust/App/Directory.pm6]: what's this?

File: lib/Crust/App/Directory.pm6 [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            'Content-Type' => "text/plain",
+            'Location' => "/{{$dir}}/"
+        ],
+        ["".encode('utf-8')] unless $path.ends-with('/');
+
+    my Str $files;
+    for $dir.IO.dir -> $file {
+        my $ct = $file.d ?? '' !! Crust::MIME.mime-type($file) || 'text/plain';
+        my $size = $file.d ?? '' !! $file.s;
+        $files ~= sprintf($dir_file, $file.basename, $file.basename, $size, $ct, DateTime.new($file.modified));
```
Comment (tokuhirom) [inline lib/Crust/App/Directory.pm6]: HTML Escape is required.

File: lib/Crust/App/Directory.pm6 [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    return
+        301,
+        [
+            'Content-Type' => "text/plain",
+            'Location' => "/{{$dir}}/"
+        ],
+        ["".encode('utf-8')] unless $path.ends-with('/');
+
+    my Str $files;
+    for $dir.IO.dir -> $file {
```
Comment (tokuhirom) [inline lib/Crust/App/Directory.pm6]: Why did you remove directory traversal defender in Plack::App::File?

File: lib/Crust/App/Directory.pm6 (hunk 2 above)
Comment (mattn) [inline lib/Crust/App/Directory.pm6]: removed

File: lib/Crust/App/Directory.pm6 (hunk 3 above)
Comment (mattn) [inline lib/Crust/App/Directory.pm6]: done
